### DISCUSSION TITLE
string/split

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#120] [Announcements]
# string/split

Java程序员进阶之路，小白的零基础Java教程，从入门到进阶，如何在Java中优雅地分割String字符串？

https://tobebetterjavaer.com/string/split.html

**Comment by CH-VictorChan** [comment]
预期连续的分割符中间也能有一个空字符串该怎么处理

**Reply by itwanger** [reply]
能给出具体的例子吗，感觉需求不够明确，什么叫预期连续？

**Reply by CH-VictorChan** [reply]
Sry 我本来想说：比如分割一个错误的IP 192.168..1 用“.”去分割的时候只能得到三位 我想得到四位该怎么做，这是以前刷一道算法题遇到的；
但是我刚刚去试了下可以分割得到四位，可能我记错了。

**Reply by clicker57** [reply]
你说的可能是这种场景："a,b,,".split(",") ---> ["a", "b"]   split方法会把切分后数组末尾的空元素都丢掉。
如果不想丢弃，需要指定split方法的第二个参数："a,b,,".split(",", -1)

**Comment by Ricehahaa** [comment]
不熟正则表达式看得一头雾水

**Comment by dcbbsp** [comment]
没有学过一遍的话这字符串部分看的有些吃力啊
